keymap/vim.js

```diff
@@ -2434,7 +2434,8 @@
               CodeMirror.commands.newlineAndIndent;
           newlineFn(cm);
         }
-        this.enterInsertMode(cm, { repeat: actionArgs.repeat }, vim);
+        var that = this;
+        setTimeout(that.enterInsertMode.bind(that, cm, { repeat: actionArgs.repeat }, vim),1);
```

Comment:
uses a time out to push off going into insert mode so all the CodeMirror operations from adding the new line have finished.

I didn't see the CodeMirror operations exposed anywhere.